index.js

```diff
@@ -96,38 +104,40 @@ class Registrar extends Emitter {
     }
   }
 
+  /**
+   * if param realm exists then returns count of users belonging to a realm,
+   * otherwise returns count of all registered users
+   * @param {String} realm - nullable realm (e.g. drachtio.org)
+   * @returns {int} count of users
+   */
   async getCountOfUsers(realm) {
-    try {
-      const users = new Set();
-      let idx = 0;
-      const pattern = makeUserPattern(realm);
-      do {
-        const res = await this.client.scan([idx, 'MATCH', pattern, 'COUNT', 100]);
-        const next = res[0];
-        const keys = res[1];
-        debug(next, keys,  `Registrar:getCountOfUsers result from scan cursor ${idx} ${realm}`);
-        keys.forEach((k) => users.add(k));
-        idx = parseInt(next);
-      } while (idx !== 0);
-      return users.size;
-    } catch (err) {
-      debug(err);
-      this.logger.error(err, 'getCountOfUsers: Error retrieving registered users');
-    }
+    const users = await this.getRegisteredUsersForRealm(realm);
+    debug(`Registrar#getCountOfUsers result=${users.length} realm=${realm}`);
+    return users.length;
   }
 
+
+  /**
+   * if realm exists then return all user parts belonging to a realm,
+   * otherwise returns all registered user parts
+   * @param {String | undefined} realm - realm (e.g. drachtio.org)
+   * @returns {[String]} Set of userParts
+   */
   async getRegisteredUsersForRealm(realm) {
     try {
+      //cleanup expired entries
+      await this.client.zremrangebyscore('active-user', 0, Date.now());
+      const keyPattern = realm ? `*${realm}` : '*';
+      const pattern = realm ? new RegExp('^(.*)@' + realm + '$') : new RegExp('^(.*)@.*$');
```

Comment:
is 'pattern' still used anywhere?  I'm probably missing it but I don't see it used below